tests/test_multipart_operations.py

```diff
@@ -4,92 +4,282 @@
+            resp["bucket_name"],
+            obj_name,
+            resp[f"{obj_name}" + "_upload_id"],
+            resp["all_part_info"],
+        )
+        log.info(mp_response)
+        log.info("Multipart operation is completed")
+
+    def test_s3_multipart_upload_part_copy(
+        self, c_scope_s3client, tmp_directories_factory
+    ):
+        """
+
+        Test multipart object list operations using BOTO s3:
+        1. Write multipart objects to the bucket
+        2. Create new bucket
+        3. Copy object data from bucket created in step 1 to new bucket
+
+        """
+        log.info("Uploading multipart object")
+        resp = self.upload_multipart_object(c_scope_s3client, tmp_directories_factory)
+        obj_name = resp["object_names"][0]
+        c_scope_s3client.complete_multipart_object_upload(
+            resp["bucket_name"],
+            obj_name,
+            resp[f"{obj_name}" + "_upload_id"],
+            resp["all_part_info"],
+        )
+        log.info("Creating another bucket to copy data")
+        new_bucket = c_scope_s3client.create_bucket()
+        log.info("Generating upload id for the multipart object")
+        get_upload_id = c_scope_s3client.initiate_multipart_object_upload(
+            new_bucket,
+            obj_name,
+        )
+        log.info("Copying data using upload_part_copy method")
+        upload_part_copy = c_scope_s3client.multipart_upload_part_copy(
+            new_bucket,
+            obj_name,
+            resp["bucket_name"] + "/" + obj_name,
+            1,
+            get_upload_id,
+        )
+        assert (
+            upload_part_copy["ResponseMetadata"]["HTTPStatusCode"] == 200
+        ), f"Failed to copy data from {resp['bucket_name']} to {new_bucket}"
+        log.info(upload_part_copy)
+        log.info("Data copied successfully from source bucket to new bucket")
+        log.info("Trying to complete multipart operation for new object")
+        all_part_info = []
+        all_part_info.append(
+            {"PartNumber": 1, "ETag": upload_part_copy["CopyPartResult"]["ETag"]}
+        )
+        c_scope_s3client.complete_multipart_object_upload(
+            new_bucket, obj_name, get_upload_id, all_part_info
+        )
+        log.info("Multipart operation is completed using upload_part_copy method")
+
+    def test_s3_multipart_abort_upload(self, c_scope_s3client, tmp_directories_factory):
+        """
+
+        Test multipart object list operations using BOTO s3:
+        1. Write multipart objects to the bucket
+        2. Abore multipart upload
+
+        """
+        log.info("Uploading multipart object")
+        resp = self.upload_multipart_object(c_scope_s3client, tmp_directories_factory)
+        log.info("Aborting Multipart operation")
+        obj_name = resp["object_names"][0]
+        abort_resp = c_scope_s3client.abort_multipart_upload(
+            resp["bucket_name"], obj_name, resp[f"{obj_name}" + "_upload_id"]
```

Comment:
f"{obj_name}" is equivalent to obj_name - see the other comment for suggestoins

Reply:
Fixed